WordPress/Classes/ViewRelated/Blog/Site Picker/SitePickerViewController+SiteIcon.swift

```diff
@@ -36,6 +33,10 @@ extension SitePickerViewController {
     }
 
     private func makeUpdateSiteIconActions() -> [UIAction] {
+        guard siteIconShouldAllowDroppedImages() else {
+            return [] // Not eligible to change the icon
+        }
```

Comment:
For context, from the issue this addresses: https://github.com/wordpress-mobile/WordPress-iOS/issues/21507#issuecomment-1710002152

> In order to fix it, I'll move the eligibility check inside UIDeferredMenuElement.uncached so that it's performed every time you invoke the menu.

Makes sense!